src/component/parser/json/converter/CategoryConverter.ts

```diff
@@ -27,8 +27,7 @@ export default class CategoryConverter {
   constructor(private convertedElements: ConvertedElements) {}
 
   deserialize(definitions: TDefinitions): void {
-    ensureIsArray<TCategory>(definitions.category).forEach(category =>
-      ensureIsArray(category.categoryValue).forEach(categoryValue => this.convertedElements.registerCategoryValue(categoryValue.id, categoryValue.value)),
-    );
+    for (const category of ensureIsArray<TCategory>(definitions.category))
+      for (const categoryValue of ensureIsArray(category.categoryValue)) this.convertedElements.registerCategoryValue(categoryValue.id, categoryValue.value);
```

Comment:
**thought:** we could probably start with a map to only keep a single for loop. I hoped this would be clearer but it is not IMHO 😄 

```suggestion
    for (const categoryValue of ensureIsArray<TCategory>(definitions.category)
      .map(category => ensureIsArray(category.categoryValue))
      .flat())
      this.convertedElements.registerCategoryValue(categoryValue.id, categoryValue.value);
```

There are other places with the same loop of loop pattern that are modified in this PR. For example, `EventDefinitionConverter.ts`.

Reply:
good catch :slightly\_smiling\_face: